tests/utils/test_feature_utils.py

```diff
@@ -56,73 +57,51 @@ def test_convert_to_homogenous_coordinates_incorrect_dim(self):
-                [-0.65653216, -1.64133041, -19.75129171],
-            ]
-        )
+        """Test conversion of valid 2D points to epipolar lines using the fundamental matrix, against manual
+        computation and with OpenCV's output."""
 
-        np.testing.assert_allclose(computed, expected)
+        points = np.array([[10.0, -5.0], [3.5, 20.0],])  # 2d points in homogenous coordinates
+        E_matrix = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0])))
+        F_matrix = E_matrix.matrix()  # using identity intrinsics
+        expected_opencv = cv.computeCorrespondEpilines(points.reshape(-1, 1, 2), 1, F_matrix)
+        expected_opencv = np.squeeze(expected_opencv)
+
+        computed = feature_utils.convert_to_epipolar_lines(points, F_matrix)
+        computed_normalized = computed / np.linalg.norm(computed[:, :2], axis=1, keepdims=True)
+        np.testing.assert_allclose(computed_normalized, expected_opencv)
 
     def test_convert_to_epipolar_lines_empty_input(self):
         """Test conversion of 0 2D points to epipolar lines using the essential matrix."""
 
         points = np.array([])  # 2d points in homogenous coordinates
-        essential_mat = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0])))
-
-        computed = feature_utils.convert_to_epipolar_lines(points, essential_mat)
+        f_matrix = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0]))).matrix()
 
+        computed = feature_utils.convert_to_epipolar_lines(points, f_matrix)
         self.assertIsNone(computed)
 
     def test_convert_to_epipolar_lines_none_input(self):
         """Test conversion of None to epipolar lines using the essential matrix."""
 
         points = None
-        essential_mat = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0])))
-
-        computed = feature_utils.convert_to_epipolar_lines(points, essential_mat)
+        E_matrix = EssentialMatrix(Rot3.RzRyRx(0, np.deg2rad(45), 0), Unit3(np.array([-5, 2, 0])))
+        F_matrix = E_matrix.matrix()  # using identity intrinsics
 
+        computed = feature_utils.convert_to_epipolar_lines(points, F_matrix)
         self.assertIsNone(computed)
 
-    def test_compute_point_line_distances(self):
-        """Test for 2D point-line distance computation."""
-
-        points = np.array(
-            [
-                [-2.0, 1.0],
-                [1.0, 3.0],
-            ]
-        )
-        lines = np.array(  # coefficients (a, b, c) for the line ax + by + c = 0
-            [
-                [1.0, 0.0, -1.0],
-                [-1.0, 1.0, 2.0],
-            ]
-        )
-        expected = np.array([3.0, 2 * math.sqrt(2)])
-        computed = feature_utils.compute_point_line_distances(points, lines)
+    def test_point_line_dotproduct(self):
+        """Test for 2D point-line dot product."""
 
+        points = np.array([[-2.0, 1.0], [1.0, 3.0],])
```

Comment:
computation would be helpful here

(1,0)
(-1,1) are quite simple

Reply:
I think because we have no norm, the computation is simple with the values I have right now too.